src/v/redpanda/admin_server.cc

```diff
@@ -408,6 +410,17 @@ ss::future<> admin_server::throw_on_error(
             throw ss::httpd::server_error_exception(
               fmt::format("Unexpected raft error: {}", ec.message()));
         }
+    } else if (ec.category() == cluster::tx_error_category()) {
+        switch (cluster::tx_errc(ec.value())) {
+        case cluster::tx_errc::leader_not_found:
+            throw co_await redirect_to_leader(req, ntp);
+        case cluster::tx_errc::pid_not_found:
+            throw ss::httpd::bad_request_exception(
+              fmt_with_ctx(fmt::format, "Can not find pid for ntp:{}", ntp));
```

Comment:
should this error be caught in a location where you can annotate the message with the pid? I suppose it is probably in the URL experiencing the error, but so is ntp. this probably doesn't matter I just thought I'd ask about it.